superset-frontend/src/explore/components/ExploreChartPanel.jsx

```diff
@@ -73,67 +178,89 @@ class ExploreChartPanel extends React.PureComponent {
           height > 0 && (
             <ChartContainer
               width={Math.floor(width)}
-              height={parseInt(this.props.height, 10) - headerHeight}
+              height={chartSectionHeight}
               annotationData={chart.annotationData}
               chartAlert={chart.chartAlert}
               chartStackTrace={chart.chartStackTrace}
               chartId={chart.id}
               chartStatus={chart.chartStatus}
-              triggerRender={this.props.triggerRender}
-              datasource={this.props.datasource}
-              errorMessage={this.props.errorMessage}
-              formData={this.props.form_data}
-              onQuery={this.props.onQuery}
-              owners={this.props?.slice?.owners}
+              triggerRender={props.triggerRender}
+              datasource={props.datasource}
+              errorMessage={props.errorMessage}
+              formData={props.form_data}
+              onQuery={props.onQuery}
+              owners={props?.slice?.owners}
               queryResponse={chart.queryResponse}
-              refreshOverlayVisible={this.props.refreshOverlayVisible}
-              setControlValue={this.props.actions.setControlValue}
-              timeout={this.props.timeout}
+              refreshOverlayVisible={props.refreshOverlayVisible}
+              setControlValue={props.actions.setControlValue}
+              timeout={props.timeout}
               triggerQuery={chart.triggerQuery}
-              vizType={this.props.vizType}
+              vizType={props.vizType}
             />
           )
         }
       </ParentSize>
     );
-  }
+  };
 
-  render() {
-    if (this.props.standalone) {
-      // dom manipulation hack to get rid of the boostrap theme's body background
-      const standaloneClass = 'background-transparent';
-      const bodyClasses = document.body.className.split(' ');
-      if (bodyClasses.indexOf(standaloneClass) === -1) {
-        document.body.className += ` ${standaloneClass}`;
-      }
-      return this.renderChart();
+  if (props.standalone) {
+    // dom manipulation hack to get rid of the boostrap theme's body background
+    const standaloneClass = 'background-transparent';
+    const bodyClasses = document.body.className.split(' ');
+    if (bodyClasses.indexOf(standaloneClass) === -1) {
```

Comment:
```suggestion
    if (!bodyClasses.includes(standaloneClass)) {
```

Reply:
OK, I admit the above suggestion, and this one, are kind of out of scope for this PR, but I'll just mention this anyway. We can potentially skip all this gross DOM and class manipulation jazz, and just use the prop to control some Emotion styles applied to the body tag with Emotions `Global` component.

Reply:
I changed `indexOf` to `includes`. As for getting rid of this "hackiness" - like you said, let's do it in a separate PR